engine/job.go

```diff
@@ -248,3 +255,15 @@ func (job *Job) StatusCode() int {
 func (job *Job) SetCloseIO(val bool) {
 	job.closeIO = val
 }
+
+// When called, causes the Job.WaitCancelled channel to unblock.
+func (job *Job) Cancel() {
+	job.cancelOnce.Do(func() {
```

Comment:
Can we get a `log.Infoln("Job canceled due to connection lost")`  type of thing here?
I was looking at the logs while I tested this and noticed that at the INFO level we print stuff about the job starting, what job it is, and it ending.  But we don't say that the engine killed it unless we turn on Debug.  For those who don't run with Debug but just INFO,I think we should be consistent and add another log line.  Then all of this info is visible at the same log level and can help figure out what actually happened.  W/o this it looks like the RUN cmd that I was on died on its own, which isn't true - the daemon killed it.